roles/splunk_common/tasks/get_facts.yml

```diff
@@ -19,23 +19,26 @@
 
 - name: "Set target version fact"
   set_fact:
-    splunk_target_version: "{{ splunk.build_location | regex_search(regexp, '\\1\\2\\3') | default('0') }}"
+    splunk_target_version: "{{ splunk.build_location | regex_search(regexp, '\\1.\\2.\\3') | default('0') }}"
   vars:
-    regexp: 'splunk(?:forwarder)?-(\d)\.(\d)\.(\d)-'
+    regexp: 'splunk(?:forwarder)?-(\d+)\.(\d+)\.(\d+)-'
+  when: "'build_location' in splunk and splunk.build_location is not none"
 
 - name: "Find manifests"
   find:
     paths: "{{ splunk.home }}"
     patterns: ".*-manifest$"
     use_regex: yes
   register: manifests
+  when: "'build_location' in splunk and splunk.build_location is not none"
```

Comment:
What is the idea to not check for the installed version, based on the fact an installer is defined or not?

Reply:
This is just an optimization, since `manifests` is only used below, for comparison to the version included in the build installer's filename.

Reply:
It is indeed used to do the comparison. But also to detect what the current version is.
This could be used later on to change certain behaviour based on versions if this would later on become necessary and this should be independent if it is freshly installed or not.

Reply:
Makes sense. I'll remove the `when`s so that `splunk_current_version` gets set.